test/e2e/util.go

```diff
@@ -1337,3 +1338,69 @@ func matchIPv4StringFamily(ipStrings []string) (string, error) {
 func matchIPv6StringFamily(ipStrings []string) (string, error) {
 	return util.MatchIPStringFamily(true /*ipv6*/, ipStrings)
 }
+
+// This is a replacement for e2epod.DeletePodWithWait(), which does not handle pods that
+// may be automatically restarted (https://issues.k8s.io/126785)
+func deletePodWithWait(ctx context.Context, c clientset.Interface, pod *v1.Pod) error {
+	if pod == nil {
+		return nil
+	}
+	if pod.UID == "" {
+		// We only recurse into deletePodWithWaitByName when UID is *not* set, to
+		// avoid infinite loops.
+		return deletePodWithWaitByName(ctx, c, pod.Name, pod.Namespace)
+	}
+	oldUID := pod.UID
+
+	framework.Logf("Deleting pod %q in namespace %q", pod.Name, pod.Namespace)
+	err := c.CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil // assume pod was already deleted
+		}
+		return fmt.Errorf("pod Delete API error: %w", err)
+	}
+	framework.Logf("Wait up to %v for pod %q to be fully deleted", e2epod.PodDeleteTimeout, pod.Name)
+	err = waitForPodNotFoundInNamespace(ctx, c, pod.Name, pod.Namespace, oldUID, e2epod.PodDeleteTimeout)
+	if err != nil {
+		return fmt.Errorf("pod %q was not deleted: %w", pod.Name, err)
+	}
+	return nil
+}
+
+// This is a replacement for e2epod.DeletePodWithWaitByName(), which does not handle pods
+// that may be automatically restarted (https://issues.k8s.io/126785)
+func deletePodWithWaitByName(ctx context.Context, c clientset.Interface, podName, podNamespace string) error {
+	pod, err := c.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil // assume pod was already deleted
+		}
+		return fmt.Errorf("pod Get API error: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("pod Get API error for %s/%s: %w", err, podNamespace, podName)
```
nit: is it worth it including that info or will the error returned have that info?

Reply:
I don't know... this is mostly a copy of the existing k8s e2e method.
It will end up being clear what pod it's talking about based on where the error gets asserted against anyway.